patches/@ethereum-waffle+compiler+3.1.0.patch

```diff
@@ -0,0 +1,32 @@
+diff --git a/node_modules/@ethereum-waffle/compiler/dist/cjs/saveOutput.js b/node_modules/@ethereum-waffle/compiler/dist/cjs/saveOutput.js
+index 602f0ee..6d53aad 100644
+--- a/node_modules/@ethereum-waffle/compiler/dist/cjs/saveOutput.js
++++ b/node_modules/@ethereum-waffle/compiler/dist/cjs/saveOutput.js
+@@ -41,10 +41,10 @@ function getMinimalContent(contractJson, config) {
+     return JSON.stringify({ abi, bytecode }, null, 2);
+ }
+ async function saveOutputSingletons(output, config, filesystem = fsOps) {
+-    for (const [, file] of Object.entries(output.contracts)) {
++    for (const [source_path, file] of Object.entries(output.contracts)) {
+         for (const [contractName, contractJson] of Object.entries(file)) {
+             const filePath = path_1.join(config.outputDirectory, `${contractName}.json`);
+-            filesystem.writeFile(filePath, getContent(contractJson, config));
++            filesystem.writeFile(filePath, getContent(contractJson, output.sources[source_path] ? output.sources[source_path].ast : undefined, source_path, contractName, 'unknown', config));
```

Comment:
There is a small feature regression here, I didn't figure out how to get the compiler version into the artifact so it just emits 'unknown'.